substrate/frame/support/src/traits/tokens/fungible/union_of.rs

```diff
@@ -0,0 +1,1005 @@
+						fungibles::Imbalance::forget(inner_credit);
+						Ok(credit)
+					},
+					Err(inner_debt) => {
+						let debt = fungibles::Imbalance::new(asset, inner_debt.peek());
+						fungibles::Imbalance::forget(inner_debt);
+						Err(debt)
+					},
+				}
+			},
+		}
+	}
+	fn withdraw(
+		asset: Self::AssetId,
+		who: &AccountId,
+		value: Self::Balance,
+		precision: Precision,
+		preservation: Preservation,
+		force: Fortitude,
+	) -> Result<fungibles::Credit<AccountId, Self>, DispatchError> {
+		match Criterion::convert(asset.clone()) {
+			Left(()) => <Left as fungible::Balanced<AccountId>>::withdraw(
+				who,
+				value,
+				precision,
+				preservation,
+				force,
+			)
+			.map(|inner_credit| {
+				let credit = fungibles::Imbalance::new(asset, inner_credit.peek());
+				fungible::Imbalance::forget(inner_credit);
+				credit
+			}),
+			Right(a) => <Right as fungibles::Balanced<AccountId>>::withdraw(
+				a,
+				who,
+				value,
+				precision,
+				preservation,
+				force,
+			)
+			.map(|inner_credit| {
+				let credit = fungibles::Imbalance::new(asset, inner_credit.peek());
+				fungibles::Imbalance::forget(inner_credit);
+				credit
+			}),
+		}
+	}
+}
+
+impl<
+		Left: fungible::BalancedHold<AccountId>,
+		Right: fungibles::BalancedHold<AccountId, Balance = Left::Balance, Reason = Left::Reason>,
+		Criterion: Convert<AssetKind, Either<(), Right::AssetId>>,
+		AssetKind: AssetId,
+		AccountId,
+	> fungibles::BalancedHold<AccountId> for UnionOf<Left, Right, Criterion, AssetKind, AccountId>
+{
+	fn slash(
+		asset: Self::AssetId,
+		reason: &Self::Reason,
+		who: &AccountId,
+		amount: Self::Balance,
+	) -> (fungibles::Credit<AccountId, Self>, Self::Balance) {
+		match Criterion::convert(asset.clone()) {
+			Left(()) => {
+				let (inner_credit, amount) =
+					<Left as fungible::BalancedHold<AccountId>>::slash(reason, who, amount);
+				let credit = fungibles::Imbalance::new(asset, inner_credit.peek());
+				fungible::Imbalance::forget(inner_credit);
+				(credit, amount)
+			},
+			Right(a) => {
+				let (inner_credit, amount) =
+					<Right as fungibles::BalancedHold<AccountId>>::slash(a, reason, who, amount);
+				let credit = fungibles::Imbalance::new(asset, inner_credit.peek());
+				fungibles::Imbalance::forget(inner_credit);
+				(credit, amount)
+			},
+		}
+	}
+}
+
+impl<
+		Left: fungible::Inspect<AccountId>,
+		Right: fungibles::Inspect<AccountId, Balance = Left::Balance> + fungibles::Create<AccountId>,
+		Criterion: Convert<AssetKind, Either<(), Right::AssetId>>,
+		AssetKind: AssetId,
+		AccountId,
+	> fungibles::Create<AccountId> for UnionOf<Left, Right, Criterion, AssetKind, AccountId>
+{
+	fn create(
+		asset: AssetKind,
+		admin: AccountId,
+		is_sufficient: bool,
+		min_balance: Self::Balance,
+	) -> DispatchResult {
+		match Criterion::convert(asset) {
+			// no-op for `Left` since `Create` trait is not defined within `fungible::*`.
+			Left(()) => Ok(()),
```

Comment:
thoughts on it returning an `Err`? seems like it should never be called on the native asset